File: giftastic-frontend-deploy/frontend/src/services/orderService.js
import api from './api';

export const orderService = {
  async placeOrder(orderData) {
    const response = await api.post('/orders', orderData);
    return response.data;
  },
  
  async placeGuestOrder(guestData) {
    const response = await api.post('/orders/guest-checkout', guestData);
    return response.data;
  },

  async getOrderById(id) {
    const response = await api.get(`/orders/${id}`);
    return response.data;
  },

  async trackGuestOrder(id, email, phone) {
    const response = await api.get(`/orders/guest-track/${id}`, {
      params: { email, phone }
    });
    return response.data;
  },

  async getCustomerOrders(customerId) {
    const response = await api.get(`/orders/customer/${customerId}`);
    return Array.isArray(response.data) ? response.data : (response.data?.content || []);
  },

  async getVendorOrders() {
    const response = await api.get('/orders/vendor');
    return response.data;
  },

  async getAllOrders() {
    const response = await api.get('/orders');
    return Array.isArray(response.data) ? response.data : (response.data?.content || []);
  },

  async updateOrderStatus(id, status) {
    const response = await api.patch(`/orders/${id}/status`, null, {
      params: { status }
    });
    return response.data;
  },

  async changePaymentMethod(id, customerId, paymentMethod, instapayPhoneNumber = null, instapayRefundPhoneNumber = null, instapayRefundName = null) {
    await api.patch(`/orders/${id}/payment-method`, {
      customerId,
      paymentMethod,
      instapayPhoneNumber,
      instapayRefundPhoneNumber,
      instapayRefundName,
    });
  },

  async submitInstapayTransactions(id, customerId, transactionIds) {
    await api.post(`/orders/${id}/instapay-transactions`, { customerId, transactionIds });
  },

  async submitGuestInstapayTransactions(id, email, phone, transactionIds) {
    await api.post(`/orders/guest-track/${id}/instapay-transactions`, { email, phone, transactionIds });
  },

  async cancelOrder(id, customerId) {
    await api.post(`/orders/${id}/cancel`, null, { params: { customerId } });
  },

  async updateVendorStatus(id, status) {
    await api.patch(`/orders/${id}/vendor-status`, { status });
  }
};

File: giftastic-deployment-env/src/main/java/com/giftastic/giftastic/modules/order/service/OrderService.java
package com.giftastic.giftastic.modules.order.service;

import java.util.List;
import java.util.UUID;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.giftastic.giftastic.modules.order.domain.GuestInfo;
import com.giftastic.giftastic.modules.order.domain.Order;
import com.giftastic.giftastic.modules.order.domain.OrderItem;
import com.giftastic.giftastic.modules.order.domain.VendorOrderStatus;
import com.giftastic.giftastic.modules.order.dto.VendorOrderResponse;

import jakarta.transaction.Transactional;

public interface OrderService {
    Order placeOrder(UUID customerId, String customerName, String customerEmail, List<OrderItem> items, String shippingAddress, String paymentMethod, String instapayPhoneNumber, String instapayRefundPhoneNumber, String instapayRefundName, UUID deliveryZoneId);
    Order placeGuestOrder(GuestInfo guestInfo, List<OrderItem> items, String paymentMethod, String instapayPhoneNumber, String instapayRefundPhoneNumber, String instapayRefundName, UUID deliveryZoneId);
    
    @Transactional
    Order checkoutCart(UUID customerId);

    @Transactional
    void cancelOrder(UUID orderId);

    void processPayment(UUID orderId);

    @Transactional
    void shipOrder(UUID orderId);

    void updateStatus(UUID orderId, String status);
    void changePaymentMethod(UUID orderId, UUID customerId, String paymentMethod, String instapayPhoneNumber, String instapayRefundPhoneNumber, String instapayRefundName);
    void submitInstapayTransactionIds(UUID orderId, UUID customerId, List<String> transactionIds);
    void submitGuestInstapayTransactionIds(UUID orderId, String email, String phone, List<String> transactionIds);
    void confirmInstapayPayment(UUID orderId, UUID adminId);
    void rejectInstapayPayment(UUID orderId, String reason);
    void updateVendorStatus(UUID orderId, UUID supplierId, VendorOrderStatus status);
    void invalidateVendorPortion(UUID orderId, UUID supplierId, UUID invalidatedBy, String reason, String details);
    
    Order getOrderById(UUID orderId);
    Order getGuestOrder(UUID orderId, String email, String phone);
    
    /**
     * Get order security information for authorization checks.
     * Returns the order ID, customer ID, and all supplier IDs of products in the order.
     */
    com.giftastic.giftastic.common.dto.OrderSecurityDTO getOrderSecurity(UUID orderId);
    
    Page<Order> getCustomerOrders(UUID customerId, Pageable pageable);
    Page<Order> getVendorOrders(UUID supplierId, Pageable pageable);
    Page<VendorOrderResponse> getVendorOrderResponses(UUID supplierId, Pageable pageable);
    Page<Order> getAllOrders(Pageable pageable);
    Page<Order> getPendingConfirmationOrders(Pageable pageable);
}
